### PR TITLE
values.yaml format changes

### PR DESCRIPTION
**Pull request by harryttd**
PR contains an overhaul of the values.yaml file. Most notable is that accounts and node have turned to maps from lists. A lot of comments have been added. Also property names were made more clear as to what they are.

This way pods and accounts can be looked up in constant time instead of always having to loop through lists.

Other changes/additions:
- templating of statefulset names. This is important for map of nodes lookups. 
- ability to configure the activation account of private chains. 
- can configure expected-proof-of-work
- The format of parameters.json is mimicked in values.yaml.
- The format of config.json is not completely mimicked, but the genesis field is. The `activation_account_name` that should reference the account that will be used to activate the chain goes here.
- Removed the chain_type field. We check what logic needs to be run in other ways now. e.g. if there is a zerotier config, it is a private chain. If the the genesis field is set in config.json, it is a private or test net.
- The  chain_name property can be both the name of your custom chain as well as the name of a public network.
- Rolling and Full snapshots url get their own fields.
- Added a flag `should_generate_unsafe_deterministic_data` that should only be used for local/private testnets. This makes spinning up chains much faster than say having to run mkchain with tezos-client to gen keys.
- Added a toggle for if commitments should be added to parameters.json. Commitments were needed for Florence net. But in general they are not used and it is a huge file that is not needed by config-generator.
- Every baker verifies that it has a secret key. Fails otherwise. Other nodes will not fail to run. They will import whatever keys are specified.
- Updated mkchain. It supports only creating a private chain for now as opposed to configuring joining a public chain. If not used, running Helm will use just the defualt values.yaml file.  It will download a rolling mainnet snapshot for one regular peer node which will connect to mainnet.

### CODE REVIEW

File: charts/tezos/templates/configs.yaml [hunk 1]
```diff
@@ -2,40 +2,41 @@ apiVersion: v1
...
-      "activation_account": "{{ (index .Values.accounts 0).name }}",
       "bootstrap_peers": {{ toJson .Values.bootstrap_peers }},
-      "chain_name": "{{ .Values.chain_name }}",
       "chain_type": "{{ .Values.chain_type }}",
-      "defualt_bootstrap_mutez": "{{ .Values.protocol.bootstrap_mutez }}",
-      "genesis_block": "{{ .Values.genesis.block }}",
-      "is_old_accounts_parameter_format": {{ default false .Values.is_old_accounts_parameter_format }},
-      "network": "{{ .Values.network }}",
-      "nodes": {{ toJson .Values.nodes }},
+      "defualt_bootstrap_mutez": "{{ .Values.bootstrap_mutez }}",
```
Comment (nicolasochem) [inline charts/tezos/templates/configs.yaml:7]: typo

Comment (harryttd) [reply charts/tezos/templates/configs.yaml:7]: good catch. Roland's PR fixes this but i can change now

**Review by nicolasochem — APPROVED** [review]
I tried the following:

* create a chain with mkchain (no params) and launch it
* create a chain with mkchain and zerotier params. Tried creating and inviting
* deploy helm with an empty values.yaml
* create a chain with mkchain with several bakers and nodes
* deterministically create accounts with mkchain arg
*  deploy florencenet node with custom values.yaml with snapshots (full and rolling)
* adjust difficulty parameter

All of the above worked according to the description.

I also carefully reviewed the default values.yaml (including comments) for intent. The current values.yaml structure is the best to satisfy all the use cases we currently support. I didn't carefully review all code changes.

**Comment by elric1** [comment]
I needed to make the following tweaks for it to work on Python 3.7 which is default on debian buster:
```
: chekalahk; git diff !(config*)
diff --git a/mkchain/MANIFEST.in b/mkchain/MANIFEST.in
index 44bda82..63bcad2 100644
--- a/mkchain/MANIFEST.in
+++ b/mkchain/MANIFEST.in
@@ -1,2 +1,3 @@
 include versioneer.py
 include tqchain/_version.py
+include tqchain/parameters.yaml
diff --git a/mkchain/tqchain/mkchain.py b/mkchain/tqchain/mkchain.py
index 8a34bb3..48cd646 100644
--- a/mkchain/tqchain/mkchain.py
+++ b/mkchain/tqchain/mkchain.py
@@ -111,11 +111,18 @@ cli_args = {
     },
 }

+class ExtendAction(argparse.Action):
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        items = getattr(namespace, self.dest) or []
+        items.extend(values)
+        setattr(namespace, self.dest, items)

 def get_args():
     parser = argparse.ArgumentParser(
         description="Generate helm values for use with the tezos-chain helm chart"
     )
+    parser.register('action', 'extend', ExtendAction)
     parser.add_argument(
         "-v",
         "--version",

```

**Comment by elric1** [comment]
I just pulled it and it appears to be working for me with the minor tweaks for Python 3.7.
I've run the code:
 - locally with minikube,
 - using an alternate docker image with minikube, and
 - in AWS via pulumi.

**Review by elric1 — APPROVED** [review]

**Pull request merged**